parlai/utils/misc.py

```diff
@@ -349,17 +350,74 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
         pass
 
 
-def nice_report(report):
+def _report_sort_key(report_key: str) -> Tuple[str, str]:
+    """
+    Sorting name for reports.
+
+    Sorts by main metric alphabetically, then by task.
+    """
+    fields = report_key.split("/")
+    main_key = fields.pop(-1)
+    sub_key = '/'.join(fields)
+    return (sub_key or 'all', main_key)
+
+
+def _float_format(f):
+    if f != f:
```

Comment:
hmm, a bit unclear - is this only to check for `None`s?

Reply:
NaNs